### PR TITLE
Remove unnecessary/deprecated promote_rule & promote_op from conversion and promotion man

### PR DESCRIPTION
**Pull request by musm**
Backport to 0.5

### CODE REVIEW

File: doc/manual/conversion-and-promotion.rst [hunk 1]
```diff
@@ -377,21 +374,18 @@ Finally, we finish off our ongoing case study of Julia's rational number
...
-result of promotion of its numerator/denominator type with the other
-integer type. The third rule applies the same logic to two different
-types of rational numbers, resulting in a rational of the promotion of
-their respective numerator/denominator types. The fourth and final rule
-dictates that promoting a rational with a float results in the same type
-as promoting the numerator/denominator type with the float.
+The first rule says that promoting a rational number with any other integer
+type promotes to a rational type whose numerator/denominator type is the
+result of promotion of its numerator/denominator type with the other integer
+type. The third rule applies the same logic to two different types of rational
```
Comment (tkelman) [inline doc/manual/conversion-and-promotion.rst]: should now say second, and "fourth and final" should now say "third and final"

**Comment by musm** [comment]
fixed

**Comment by musm** [comment]
bump

**Comment by tkelman** [comment]
@stevengj @pabloferz please review

**Review by p-zubieta — APPROVED** [review]
LGTM

**Pull request merged**